### PR TITLE
Update Pins on MKS sBase

### PR DESCRIPTION
**Pull request by forkoz**
### Description

Fix invalid pin and add some comments.

### Benefits

Easier identification of usable pins and some pin capabilities.

### Related Issues

Don't be me and share end stops with the LCD.

### CODE REVIEW

File: Marlin/src/pins/pins_MKS_SBASE.h [hunk 1]
```diff
@@ -104,14 +106,16 @@
...
+// Using them with an LCD connected or configured will lead to hangs & crashes.
 //
 
 // 5V
 // NC
 // GND
 #define PIN_P0_17          P0_17
 #define PIN_P0_16          P0_16
-#define PIN_P0_14          P0_14
+#define PIN_P0_14          P0_15
```
Comment (gloomyandy) [inline Marlin/src/pins/pins_MKS_SBASE.h]: This definition seems a little strange. Should the define now read PIN_P0_15?

File: Marlin/src/pins/pins_MKS_SBASE.h [hunk 2]
```diff
@@ -104,14 +106,16 @@
 
 //
 // Connector J7
+// Note: These pins are all shared with the EXP1/EXP2 Connector.
+// Using them with an LCD connected or configured will lead to hangs & crashes.
```
Comment (gloomyandy) [inline Marlin/src/pins/pins_MKS_SBASE.h]: May be worth noting that although the cpu pins are shared, the J7 and EXP1 pins are not directly connected. The pins on EXP1 go via a level shifter (as do many of the pins on EXP1/EXP2).

File: Marlin/src/pins/pins_MKS_SBASE.h (hunk 1 above)
Comment (forkoz) [reply Marlin/src/pins/pins_MKS_SBASE.h]: good catch

File: Marlin/src/pins/pins_MKS_SBASE.h (hunk 2 above)
Comment (forkoz) [reply Marlin/src/pins/pins_MKS_SBASE.h]: What's a better way to put it?

Comment (gloomyandy) [reply Marlin/src/pins/pins_MKS_SBASE.h]: I think what you have is fine. Perhaps just add. "The connection to EXP1 goes via a level shifter."

I was a little confused by your original comment as I'd previously tested the J7 pins with a meter to see if they were shared with any other external connections and they did not seem to be. It was only when digging deeper that I noticed there is a level shifter on the EXP1 connection which means they do not show as shared with a simple meter check. Might save someone else the same confusion.

Comment (forkoz) [reply Marlin/src/pins/pins_MKS_SBASE.h]: added digitally, if that doesn't make enough sense I'll just copy the above.

Comment (gloomyandy) [reply Marlin/src/pins/pins_MKS_SBASE.h]: No that's fine!

**Pull request merged**